### PR TITLE
zqd: Simplify zqd pcap tests

### PR DESCRIPTION
**Pull request by mattnibs**
Now that we have system test coverage for zqd running with the zeek
binary, the zqd pcap handler tests can be reformatted and run w/out
a dependency on the zeekrunner. Remove and simplify these tests.

Included:
- remove `test-zeek` from `Makefile`
- Allow all pcap handler tests to run on windows.
- Rely on testing.T.Cleanup for core cleanup.

### CODE REVIEW

**Review by henridf — COMMENTED** [review]
I like the simplification, but if I'm following correctly, it seems like this change is taking us further away than we currently are from having windows Zeek (and soon by extension Suricata) exercised in zq CI, since the script-style system tests don't run on windows. (I realize that `handlers_zeek_test.go` currently `Skip()`s tests when running on Windows, but that seems like it dates back to pre-Windows days and something we ought to remove).

Or maybe we're ok with the coverage downstream integration tests doing that, though that seems a little removed. Curious to also hear what @nwt and @alfred-landrum think.

**Comment by mattnibs** [comment]
> I like the simplification, but if I'm following correctly, it seems like this change is taking us further away than we currently are from having windows Zeek (and soon by extension Suricata) exercised in zq CI, since the script-style system tests don't run on windows. (I realize that `handlers_zeek_test.go` currently `Skip()`s tests when running on Windows, but that seems like it dates back to pre-Windows days and something we ought to remove).

@henridf that's a good point. We could add windows zeek coverage by removing the windows skip. If that's the approach we want to take, I would suggest keeping most the modifications this pr, but adding back the `make test-zeek` command as well as the tag for the `handers_pcap_test.go` file.

My preferred approach is to keep this as a non-binary dependent test, and add windows capability for `make test-system`. This would require some additional plumbing to `ztest` that would allow us to specify alternative scripts windows, but it would be nice to have system test coverage on windows.

File: zqd/handlers_test.go [hunk 1]
```diff
@@ -1173,7 +1144,7 @@ func postSpaceLogs(t *testing.T, c *api.Connection, spaceID api.SpaceID, tc *ndj
 	return payloads
 }
 
-func testZeekLauncher(start, wait procFn) pcapanalyzer.Launcher {
+func testLauncher(start, wait procFn) pcapanalyzer.Launcher {
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply zqd/handlers_test.go:1147]: good idea

**Comment by henridf** [comment]
> I would suggest keeping most the modifications this pr, but adding back the make test-zeek command as well as the tag for the handers_pcap_test.go file.

That approach sounds 👍  to me.

> My preferred approach is to keep this as a non-binary dependent test, and add windows capability for make test-system.

Such coverage would be great to have. My impression is that this is a non-trivial effort, but that may be due to my not really understanding what it would take...

File: zqd/api/stream.go [hunk 2]
```diff
@@ -2,13 +2,27 @@ package api
...
 	"bufio"
+	"errors"
 	"io"
 )
 
 type Stream struct {
 	scanner *bufio.Scanner
 }
 
+type Payloads []interface{}
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply zqd/api/stream.go:13]: Indirectly. It provides the pcap test change a more ergonomic way to differentiate between http protocol errors and errors received during ingest.

**Comment by mattnibs** [comment]
Closing in a favor of an upcoming slimmed down pr.

**Pull request closed without merging**